docs/upgrade/3.0.md

```diff
@@ -0,0 +1,35 @@
+# Upgrading to 3.0
+
+## Introduction
+
+This document lists the steps necessary to upgrade from 2.x to 3.0.
+
+The full list of changes can be found in the [Changelog](https://github.com/florimondmanca/djangorestframework-api-key/blob/master/CHANGELOG.md).
+
+## Steps
+
+### 1. Upgrade your Python to 3.8+
+
+The 3.0 release drops support for Python 3.7, which has reached EOL in June 2023.
+
+Before upgrading, make sure you are running on Python 3.8 or above.
+
+### 2. Review usage of `.has_object_permission()` on DRF 3.14+ with custom API key models
+
+An implementation of `.has_object_permission()` on `BaseHasAPIKey` that was redundant with `.has_permission()` has been dropped when using Django REST Framework 3.14.0 and above.
+
+If you are using DRF 3.14+ and have custom API key models, you will want to review any implementation of `.has_object_permission()`. Calls to `super().has_object_permission()` will now return `True` (the DRF default) instead of re-validating the API key. This is the desirable thing to do, but you may need some adjustements if you relied on that somehow until now.
```

Comment:
```suggestion
If you are using DRF 3.14+ and have custom API key models, you will want to review any implementation of `.has_object_permission()`. Calls to `super().has_object_permission()` will now return `True` (the DRF default) instead of re-validating the API key. This is the desirable thing to do, but you may need some adjustements if you relied on the previous behavior somehow until now.
```